lib/dependabot/file_updaters/java_script/npm_and_yarn.rb

```diff
@@ -10,6 +10,16 @@ class NpmAndYarn < Dependabot::FileUpdaters::Base
         require_relative "npm_and_yarn/npm_lockfile_updater"
         require_relative "npm_and_yarn/yarn_lockfile_updater"
 
+        def initialize(dependencies:, dependency_files:, credentials:)
```

Comment:
I don't love the initializer doing this extra work. Would a `filtered_dependency_files` method work instead?